src/Compilers/VisualBasic/Test/CommandLine/CommandLineTests.vb

```diff
@@ -228,18 +231,36 @@ End Class
             Dim exitCode = cmd.Run(output, Nothing)
 
             Assert.Equal(0, exitCode)
+            Dim patched As String = Regex.Replace(output.ToString().Trim(), "version \d+\.\d+\.\d+(\.\d+)?", "version A.B.C.D")
+            patched = ReplaceCommitHash(patched)
             Assert.Equal(<text>
-Microsoft (R) Visual Basic Compiler version A.B.C.D
+Microsoft (R) Visual Basic Compiler version A.B.C.D (HASH)
 Copyright (C) Microsoft Corporation. All rights reserved.
 </text>.Value.Replace(vbLf, vbCrLf).Trim,
-                Regex.Replace(output.ToString().Trim(), "version \d+\.\d+\.\d+(\.\d+)?", "version A.B.C.D"))
+                patched)
             ' Privately queued builds have 3-part version numbers instead of 4.  Since we're throwing away the version number,
             ' making the last part optional will fix this.
 
-
             CleanupAllGeneratedFiles(src)
         End Sub
 
+        <Theory,
+            InlineData("Microsoft (R) Visual Basic Compiler version A.B.C.D (<developer build>)",
+                       "Microsoft (R) Visual Basic Compiler version A.B.C.D (HASH)"),
+            InlineData("Microsoft (R) Visual Basic Compiler version A.B.C.D (ABCDEF01)",
+                       "Microsoft (R) Visual Basic Compiler version A.B.C.D (HASH)"),
+            InlineData("Microsoft (R) Visual Basic Compiler version A.B.C.D (abcdef90)",
+                       "Microsoft (R) Visual Basic Compiler version A.B.C.D (HASH)"),
+            InlineData("Microsoft (R) Visual Basic Compiler version A.B.C.D (12345678)",
+                       "Microsoft (R) Visual Basic Compiler version A.B.C.D (HASH)")>
+        Public Sub TestReplaceCommitHash(orig As String, expected As String)
+            Assert.Equal(expected, ReplaceCommitHash(orig))
+        End Sub
+
+        Private Shared Function ReplaceCommitHash(s As String) As String
+            Return Regex.Replace(s, "(\((<developer build>|[a-fA-F0-9]{8})\))", "(HASH)")
+        End Function
```

Comment:
`ReplaceCommitHash` can be moved to `CommonTestBase`. Then one copy of `TestReplaceCommitHash` can be removed.